api/query.py

```diff
@@ -40,13 +50,45 @@ class Position(NamedTuple):
     line: int
     column: int
 
+    def __eq__(self, other) -> bool:
+        return self.line == other.line and other.column == self.column
+
+    def __gt__(self, other) -> bool:
+        return self.line > other.line or (
+            other.line == self.line and other.column <= self.column
+        )
+
+    def __lt__(self, other) -> bool:
+        return self.line < other.line or (
+            other.line == self.line and other.column >= self.column
+        )
+
 
 class Location(NamedTuple):
     path: str
     start: Position
     stop: Position
 
 
+class Type(NamedTuple):
+    location: Dict[str, Position]
+    annotation: str
+
+    def extract_function_model(self) -> str:
+        functions = re.findall(r"(?<=typing.Callable\().*?(?=\))", self.annotation)
```

Comment:
I think we can do better than a regex here. `pyre query "defines(package.module)"` will return you structured data with function names and parameters. That's going to be much more robust compared to using regex to pull data from a type annotation

Reply:
Hmm, definitely agree that regex doesn't seem like the cleanest solution, but I'm not sure how we can determine what we want to extract from `pyre query "defines(package.module)` without using regex on at least the annotation returned from the `types` query.

Reply:
Fair point. We could still avoid the second regex using `defines`, but I think you're right we can't avoid the first regex with the way things are currently written.